### ISSUE TITLE
`undefined` parameter as value in `findFirst` will return first row in db

### ISSUE THREAD

**Issue by davidheineback** [prisma/prisma#17723]
### Bug description

undefined parameter as value for "where" in findFirst query will return first row in i queried table instead of null/undefined

### How to reproduce

Set entityNumber to undefined

`const entityNumber = undefined`

Query the entity table with findFirst using where entity_number: entityNumber

```
const entity = await prisma.entity.findFirst({
    where: {
     entity_number: entityNumber
    }
  })
```

### Expected behavior

Expected to return undefined or null 

### Prisma information

Part of schema:

```
generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "rhel-openssl-1.0.x"]
}

datasource db {
  provider     = "mysql"
  url          = env("DATABASE_URL") // Planetscale as DBprovider.
  relationMode = "prisma"
}

model entity {
  id                  Int                 @id @unique(map: "id") @default(autoincrement())
  entity_number       String?             @db.VarChar(50)
  name                String?             @db.VarChar(50)
}
```

Query:
(entityNumber is parameter provided by a user)

```
const entity = await prisma.entity.findFirst({
    where: {
     entity_number: entityNumber
    }
  })
```

### Environment & setup

- OS: MacOS
- Database: PlanetScale (mysql)
- Node.js version: v14.2.0

### Prisma Version

´´´
"@prisma/client": "^4.8.1", 
 "prisma": "^4.8.1",
´´´

**Comment by millsp** [comment]
Hey @davidheineback, when we encounter `undefined`, we strip it away from the query and treat it as if you never passed that information.

```ts
const entity = await prisma.entity.findFirst({
  where: {
    entity_number: undefined
  }
})
```

becomes

```ts
const entity = await prisma.entity.findFirst({
  where: {
  }
})
```

**Comment by davidheineback** [comment]
@millsp That explains the result and I can understand this logic in the "data" block, however in there "where" block for me at least this a bad behaviour. Is it possible to separate the handling of undefined dependent on what block it is used in or is there a good reason for undefined to remain unhandled in  "where" ?

**Comment by millsp** [comment]
Changing that wouldn't be easy, as that would probably change the behavior in many projects. That would be a huge breaking change. I understand that it can be inconvenient and maybe even dangerous in the case of a deletion. If you want to take the time, we would appreciate your proposal/ideas on the topic.

**Comment by davidheineback** [comment]
Yes that would be a good enough solution that instead of returning first row it will return empty array.  And the same behaviour should be implemented for update and delete (as you mentioned these are the most dangerous operations with the current behavior

**Comment by vajdagabor** [comment]
I've just run into this issue… It can be pretty dangerous, especially because the behaviour is misleading this way:

```js
async function getUser(where) {
  return await prisma.user.findFirst({ where })
}

//… somewhere else in the code:

const userId = nonDefinedId
const user = getUser({ id: userId })  // oops… getting first user accidentally
if (!user) return 
signInWith(user)
charge(user)
delete(user)
```

**Comment by aryzle** [comment]
agreed with @vajdagabor and @millsp, it is _really_ weird behavior to return a row of data on a `find` when the thing you want to match on is undefined/null, as an ORM how can you even guess what the user intended? isn't it better to throw an error as a design choice? plus why would someone pass in undefined/null on purpose? The error would help point out a bug

**Comment by jharrell** [comment]
Hi folks, this may be resolved by our introduction of the [`strictUndefinedChecks` Preview feature](https://www.prisma.io/docs/orm/prisma-client/special-fields-and-types/null-and-undefined) please let me know if this makes more sense for your uses!

**Comment by ludralph** [comment]
Closing this in favour of [11021](https://github.com/prisma/prisma/issues/11021)
